string type
string client
---
PartInfo[] parts_info
